source/cpu/cpu_power/cpupower_amd64.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cpu_power
+
+type CpuidRet struct {
+	EAX, EBX, ECX, EDX uint32
+}
+
+func cpuid_low(eax, ecx uint32) CpuidRet {
+	r := CpuidRet{}
+	r.EAX, r.EBX, r.ECX, r.EDX = cpuidAsm(eax, ecx)
+	return r
+}
+
+func cpuidAsm(eax_arg, ecx_arg uint32) (eax, ebx, ecx, edx uint32)
+
+const (
+	// CPUID EAX input values
+	PROCESSOR_FREQ_INFO_ENUMERATION_LEAF = 0x16
+)
+
+func DiscoverBaseFreq() uint32 {
+	freqInfo := cpuid_low(PROCESSOR_FREQ_INFO_ENUMERATION_LEAF, 0)
+	return freqInfo.EAX
+}
```

Comment:
Please do not copy paste cpuid functionality that already exists in `source/rdt/`. But, make it common, instead, and use it from rdt and from cpu-power. That is, move the cpuid code from `rdt` to `pkg/cpuid`, for example.

Reply:
+1, will change it.